src/client/components/Profile/ProfileDetails.js

```diff
@@ -1,49 +1,158 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import './Profile.styling.css';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faPhoneSquare, faEnvelope } from '@fortawesome/fontawesome-free-solid';
-
-const ProfileDetails = ({ profileImage, userName, email }) => (
-  <>
-    <section className="name-image-container">
-      <img src={profileImage} className="profile-img" alt="users profile" />
-      <h2>{userName}</h2>
-      <p>Professional role</p>
-    </section>
-    <section className="users-information-container">
-      <div className="user-email">
-        <div className="icon">
-          <FontAwesomeIcon icon={faEnvelope} />
+import {
+  faPhoneSquare,
+  faEnvelope,
+  faCamera,
+} from '@fortawesome/fontawesome-free-solid';
+import { useStorage } from '../../hooks/useStorage';
+
+function ProfileDetails({
+  profileImage,
+  email,
+  userName,
+  phoneNumber,
+  handleSubmit,
+  buttonText,
+}) {
+  const [editMode, setEditMode] = useState(false);
+  const [formDetails, setFormDetails] = useState({
+    profileImage,
+    userName,
+    phoneNumber,
+  });
+
+  const [file, setFile] = useState(null);
+  const [error, setError] = useState(null);
+
+  const types = ['image/png', 'image/jpeg', 'image/jpg'];
+
+  const handleImageChange = (e) => {
+    const selectedFile = e.target.files[0];
+
+    if (selectedFile) {
+      if (types.includes(selectedFile.type)) {
+        setError(null);
+        setFile(selectedFile);
+        // setFormDetails.profileImage(url);
+      } else {
+        setFile(null);
+        setError('Please select an image file (png or jpg)');
+      }
+    }
+  };
+
+  const { url } = useStorage(file);
+
+  function handleEditMode() {
+    if (editMode) {
```

Comment:
Ahh so the editmode is true meaning that we have clicke the button when it was true, wanting it to be false. This could you a comment!

Basically what youa re doing here is saving the information the user wrote and editmode to false

Reply:
Implemented the above changes and awaiting for review.